124-sidecar-addition-proposal.md

```diff
@@ -0,0 +1,194 @@
+# Proposal: Native Sidecar Container Support for Strimzi Components
+
+## Summary
+Provide a brief summary of the feature you are proposing to add to Strimzi.
+
+This proposal adds first‑class, **declarative sidecar container** support to Strimzi-managed workloads (starting with Kafka brokers), configured via CRDs under `spec.kafka.template.pod.sidecarContainers`. Users can attach monitoring, logging, security, and networking sidecars without webhooks or forks. The design introduces a stable CRD schema (`SidecarContainer`), a validation and conversion pipeline, and generic interfaces so other Strimzi components (KafkaConnect, KafkaBridge, MirrorMaker2, etc.) can adopt it.
+
+## Current situation
+Describe the current capability Strimzi has in this area.
+
+Strimzi currently lacks native sidecar support. Users rely on:
+1. Mutating webhooks/admission controllers
+2. Forking and patching Strimzi code
+3. Ephemeral containers for ad‑hoc troubleshooting
+
+Limitations of these approaches include fragile upgrades, higher operational complexity, lack of declarative control, and weak integration with Strimzi lifecycle and security primitives.
+
+## Motivation
+Explain the motivation why this should be added, and what value it brings.
+
+**Business value**
+- Operational flexibility: Enable Flexibility for adding additional sidecars as per the custom need of the users
+- Simplified ops: manage sidecars natively via Strimzi CRDs and GitOps
+- Cloud‑native patterns: support log forwarders, APM agents, service‑mesh proxies
+
+**Representative use cases**
+- Monitoring/Observability: prometheus exporters, APM agents (Datadog, New Relic)
+- Logging/Auditing: Fluent Bit/Filebeat; audit collectors
+- Security/Compliance: Vault agent; cert rotation helpers
+- Networking: traffic analyzers/proxies
+- Data tooling: lightweight data checks or protocol helpers
```

Comment:
TBH, I do not thikn sidecar is a good solution for any of these. And integrating them with Strimzi this way might be either super hard or not work at all.